__tests__/user-details/task-duedate.test.js

```diff
@@ -0,0 +1,120 @@
+const puppeteer = require('puppeteer');
+const {
+  userDetailsApi,
+} = require('../../mock-data/tasks-card-date-time-end-date-user/index'); //has user info
+const {
+  superUserDetails,
+} = require('../../mock-data/tasks-card-date-time-end-date-self/index'); // has super user info
+
+describe('Tasks On User Management Page', () => {
+  let browser;
+  let page;
+
+  jest.setTimeout(60000);
+
+  beforeAll(async () => {
+    browser = await puppeteer.launch({
+      headless: 'new', //change headless to 'new' to check the tests in browser
+      ignoreHTTPSErrors: true,
+      args: ['--incognito', '--disable-web-security'],
+      devtools: false,
+    });
+    page = await browser.newPage();
+
+    await page.setRequestInterception(true);
+
+    page.on('request', (interceptedRequest) => {
+      const url = interceptedRequest.url();
+      if (url === 'https://api.realdevsquad.com/tasks/sunny-s') {
+        // When we encounter the respective api call we respond with the below response
+        interceptedRequest.respond({
+          status: 200,
+          contentType: 'application/json',
+          headers: {
+            'Access-Control-Allow-Origin': '*',
+            'Access-Control-Allow-Methods': 'GET, POST, PUT, DELETE, OPTIONS',
+            'Access-Control-Allow-Headers': 'Content-Type, Authorization',
+          },
+          body: JSON.stringify(userDetailsApi),
+        });
+      } else if (url === 'https://api.realdevsquad.com/users/self') {
+        // When we encounter the respective api call we respond with the below response
+        interceptedRequest.respond({
+          status: 200,
+          contentType: 'application/json',
+          headers: {
+            'Access-Control-Allow-Origin': '*',
+            'Access-Control-Allow-Methods': 'GET, POST, PUT, DELETE, OPTIONS',
+            'Access-Control-Allow-Headers': 'Content-Type, Authorization',
+          },
+          body: JSON.stringify(superUserDetails), // Y contains the json of a superuser in the server which will grant us the access to view the page without locks
+        });
+      } else {
+        interceptedRequest.continue();
+      }
+    });
+    await page.goto(
+      'http://localhost:8000/users/details/index.html?username=sunny-s',
+    );
+
+    await page.evaluate(async () => {
+      // We write the function with superUser as true
+
+      async function accessingUserData() {
+        const isSuperUser = true;
+        if (isSuperUser) {
+          getUserTasks();
```

Comment:
what do you think about below code
if (isSuperUser) {
    await getUserTasks();
    await getUserPrs();
    await generateAcademicTabDetails();
    toggleAccordionTabsVisibility();
  } else {
    lockAccordiansForNonSuperUser();
  }

Reply:
Hello,

I see you have used the await keyword, it means that the functions will run one after the other, whereas in my case, all of them will run concurrently without waiting.
This is my first time dealing with async and await and also writing tests. 
Please let me know whether the comment you have left for me is one of the best practices used. I will implement them in my code.

Reply:
![image](https://github.com/Real-Dev-Squad/website-dashboard/assets/31709147/9a9ce24a-a7a5-4714-bb28-36a947cc60ec)
Updated the code, tests passing.